typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/helpers/findTargetItem.ts

```diff
@@ -6,35 +6,53 @@
  * %%
  * Build Tools for VMware Aria
  * Copyright 2023 VMware, Inc.
- * 
+ *
  * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.
- * 
+ *
  * This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
  * #L%
  */
 import { WorkflowItemDescriptor } from "../../../../decorators";
 import { findItemByName } from "./findItemByName";
 
+export const DEFAULT_END_ITEM_NAME = "item0";
+
 /**
  * Helper function to find the target item for the given item
  *
- * If the `item.target` is "end", it will return "item0"
- * If the `item.target` is not null, it will return the item with the given name, or 0 if not found
- * If the `item.target` is null, it will return the next item in the workflow, or "item0" if it is the last item
- *
- *
- * @param item The item to find the target for
- * @param pos The position of the item in the workflow
- * @param items The list of all items in the workflow
- * @returns The name of the target item
+ * @param target The name of the target item (optional)
+ * @param pos The position of the current item in the workflow.
+ *  If target isn't specified, the element in the next position will be targeted.
+ * @param item The current item. Used to get the list of all items in the workflow.
+ * @returns The name of the target item:
+ * If the `item.target` is "end", it will return {@link DEFAULT_END_ITEM_NAME}
+ * If the `item.target` is defined, it will return the item with the given name, throw if not found
+ * If the `item.target` is null/undefined, it will return the next item in the workflow
+ * or {@link DEFAULT_END_ITEM_NAME} when there is no such item or it cannot be targeted
+ * If the resulting element cannot be targeted:
+ * - if explicitly targeted (target != null), logs a WARNING before returning the resulting element.
+ * - if not explicitly targeted, returns {@link DEFAULT_END_ITEM_NAME} instead and logs a corresponding DEBUG level message;
+ * @throws Error if the specified target is missing.
  */
 export function findTargetItem(target: any, pos: number, item: WorkflowItemDescriptor): string {
-	if (target === "end") {
-		return "item0";
+	const wfItems = item.parent.items;
+	const nextInd = target == null ? pos + 1 : findItemByName(wfItems, target);
+	const nextItem = wfItems[nextInd - 1];
+	const isExplicitTarget = target != null && target !== "end";
+	if (!nextItem) {
+		if (!isExplicitTarget) {
```

Comment:
Having the possibility to throw here changes the default behavior. This is something we need to be really, really careful about, as we may end up being bombarded with a bunch of issues that we changed this. Realistically the way I see it.. if there is not nextItem, point to DEFAULT_END_ITEM_NAME

Reply:
An error is thrown only if the target was set explicitly and is not "end", and should not affect decorators with no target specified (targets were introduced recently). IMO validating explicitly set targets is preferable to silently redirecting them to "item0" on a typo.

Reply:
True, we can make an exception as you say things were changed recently. This however sets a bad precedent.

Introducing a feature means that we've committed to that feature and functionality. Changing it often, means the state of "stable" releases is not so stable after all. Semantically thinking, these changes will lead to a MAJOR version change. Unless something is a bug we 100% need to make this 100% backwards compatible. Exception is if something was discussed in a triaging session, but nobody should make these assumptions alone.

I implore you, check your code for other such backwards incompatible changes as they can be problematic